lib/kube_link_generator.rb

```diff
@@ -38,16 +38,30 @@ def pod_index(name)
     index.chars.map { |c| chars.index(c) }.reduce(0) { |v, c| v * chars.length + c }
   end
 
-  def _get_pods_for_role(role_name)
-    client.get_pods(namespace: namespace, label_selector: "app.kubernetes.io/component=#{role_name}")
+  def _get_sts_image_for_role(role_name)
+    @client_stateful_set.get_stateful_set(role_name, namespace)
+      .spec.template.spec.containers.map(&:image).sort.join("\n")
+  end
+
+  def _get_pods_for_role(role_name, sts_image)
+    client
+      .get_pods(namespace: namespace, label_selector: "app.kubernetes.io/component=#{role_name}")
+      .select do |pod|
+      pod_image = pod.spec.containers.map(&:image).sort.join("\n")
+      next true if pod_image == sts_image
+
+      false
```

Comment:
You don't need `next true if … false`; just do the condition directly.

```ruby
client
  .get_pods(namespace: namespace, label_selector: "app.kubernetes.io/component=#{role_name}")
  .select { |pod| pod.spec.containers.map(&:image).sort.join("\n") == sts_image }
```

Of course, this may be clearer with more rewriting…
```ruby
def _get_image_for_pod_spec(pod_spec)
  pod_spec.containers.map(&:image).sort.join("\n")
end

def _get_sts_image_for_role(role_name)
  statefulset = @client_stateful_set.get_stateful_set(role_name, namespace)
  _get_image_for_pod_spec statefulset.spec.template.spec
end

def _get_pods_for_role(role_name, sts_image)
  client
    .get_pods(namespace: namespace, label_selector: "app.kubernetes.io/component=#{role_name}")
    .select { |pod| _get_image_for_pod_spec(pod.spec) == sts_image }
end
```

Reply:
will update

Reply:
Done with commit 9f5818a58bc40cb911cbd4b8f71ed8f481cf6e1c